vllm/v1/core/encoder_cache_manager.py

```diff
@@ -48,14 +49,20 @@ class EncoderCacheManager:
                This is cleared after every call to get_freed_ids().
     """
 
-    def __init__(self, cache_size: int):
+    def __init__(self, cache_size: int, is_encoder_decoder: bool):
```

Comment:
I am still digging in, but why is this special case needed? Shouldn't we be able to re-use this encoder cache? I understand its not explicitly needed since we must schedule all the encoder tokens at once, but perhaps this would simplify the implementation and branching if we did not have this if...else

Reply:
The encoder cache is used for referencing encoder output potentially more than once during model execution. Whisper is a bit simpler, where the encoder output is processed only once in all cases.

There may be value in caching encoder output for use across requests, but that's not the same logic implemented in the encoder cache right now, and I decided that was best left as a follow-up optimization.

I looked at using the encoder cache anyway in case it helped clean up the code to reduce special cases, but it really didn't. The processing just isn't the same as the other multimodal models.